plugins/module_utils/openssh/backends/common.py

```diff
@@ -18,25 +18,289 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
+import abc
 import os
+import stat
 
+from ansible.module_utils import six
 
-def restore_on_failure(f):
-    def backup_and_restore(module, path, *args, **kwargs):
-        backup_file = module.backup_local(path) if os.path.exists(path) else None
+from ansible_collections.community.crypto.plugins.module_utils.openssh.utils import (
+    parse_openssh_version,
+)
+
+
+def _restore_on_failure(f):
+    def backup_and_restore(self, sources_and_destinations, *args, **kwargs):
```

Comment:
`restore_on_failure` is part of the public API of this collection (even though probably nobody uses it), so these changes are breaking ones.

How about keeping `restore_on_failure` without underscore, and adding something like
```
if not isinstance(sources_and_destinations, (list, tuple)):
    sources_and_destinations = [(sources_and_destinations, sources_and_destinations)]
```

Reply:
Hmmm, one thing I missed is that previously, the first parameter was `module`, while now the module is `self.module`. I'm not sure what's the best way to fix this...

Reply:
I'll just revert `restore_on_failure` and create a new decorator for the desired functionality.